cli/azd/test/functional/vs_server_test.go

```diff
@@ -10,20 +10,133 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
+	"net/http"
+	"net/http/httptest"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/azure/azure-dev/cli/azd/pkg/contracts"
 	"github.com/azure/azure-dev/cli/azd/test/azdcli"
 	"github.com/azure/azure-dev/cli/azd/test/ostest"
 	"github.com/azure/azure-dev/cli/azd/test/recording"
+	"github.com/gorilla/websocket"
 	"github.com/stretchr/testify/require"
+	"go.lsp.dev/jsonrpc2"
 )
 
+func Test_CLI_VsServerExternalAuth(t *testing.T) {
```

Comment:
@weikanglim - I thought about trying to implement the server inside `AcceptanceTests.cs` and arrange for things to use it - but I was unsure exactly how I would implement it (I assume I would have it just call the `AzureDeveloperCliCredential` type that we added - in a very strange "snake eating its tail" sort of way) but I was a little nervous about it and felt like this test gave us most of the coverage we care about.

Reply:
@ellismg, I much prefer what you have here! Writing the smaller test, and especially when it can be written natively in `go` is much, much more preferable.

The `AcceptanceTests.cs` is largely to capture the full client -> server request flow, and especially from a .NET StreamJsonRpc client so that we can fully understand and test the full integration behavior. My thinking has always been that this grants us the highest fidelity of testing that we can ask for -- no surprise gotchas in implementing the server-side of the spec, but finding out the .NET client behaves differently.

If we are interested still in making sure the .NET part behaves correctly (which I don't think is necessary), I'd spin up the mock auth server, and pass it as a test input parameter (env var) to the .NET test similar to what you are doing here.

Reply:
> If we are interested still in making sure the .NET part behaves correctly (which I don't think is necessary), I'd spin up the mock auth server, and pass it as a test input parameter (env var) to the .NET test similar to what you are doing here.

Thanks for suggesting this idea of starting the server in Go and then just shuttling the connection information to the test and having it pass it in - for some reason, I would have never thought to do that and would have just implemented the server in the test directly.  That is a much cleaner idea.  But I'm happy not doing any of it if you are happy not doing any of it right now unless we find that we think it would really buy us something.

Reply:
No, I don't think it buys us anything, let's keep what we have.